pdf/core/utils.go

```diff
@@ -38,6 +38,18 @@ func getUniDocVersion() string {
 	return common.Version
 }
 
+// GetObjectNums returns the object numbers of the PDF objects in the file
+func (this *PdfParser) GetObjectNums() (map[int]int, error) {
```

Comment:
After reconsideration, I think this approach is good.  However, the input to pdfReader.GetIndirectObjectByNumber(key) should actually be the object number (22 in this case).  Thus a simple GetObjectNums() -> []int should be fine.

Did you ever experience a case where the object number was not the key?